### PR TITLE
move example's setup to setup.py

### PR DESCRIPTION
**Pull request by tanapiyo**
move the list of dependences for examples to setup.py

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -224,8 +224,8 @@ jobs:
           name: install
           command: |
             . venv/bin/activate
-            pip install chainer keras catboost lightgbm scikit-learn tensorflow mxnet xgboost torch torchvision dask-ml dask[dataframe]
-
+            pip install $(ls dist/*.tar.gz)[example]
+
```
Comment (sile) [inline .circleci/config.yml]: Please remove this whitespaces.
```suggestion

```

File: setup.py [hunk 2]
```diff
@@ -104,5 +109,6 @@ def find_any_distribution(pkgs):
     },
     install_requires=get_install_requires(),
     tests_require=get_extras_require()['testing'],
+    examples_require=get_extras_require()['example'],
```
Comment (sile) [inline setup.py]: I think this line can be removed.
```suggestion
```

**Comment by sile** [comment]
NOTE: This resolves #515

File: .circleci/config.yml [hunk 3]
```diff
@@ -224,8 +224,7 @@ jobs:
           name: install
           command: |
             . venv/bin/activate
-            pip install chainer keras catboost lightgbm scikit-learn tensorflow mxnet xgboost torch torchvision dask-ml dask[dataframe]
-
+            pip install $(ls dist/*.tar.gz)[example]
```
Comment (toshihikoyanase) [inline .circleci/config.yml]: Please add a blank line at the end of the `run` step.

**Pull request merged**